### ISSUE TITLE
Build script does not work on M2 Mac

### ISSUE THREAD

**Issue by D4M13N-D3V** [LadybirdBrowser/ladybird#1610]
Here is my system information.

```
    System Software Overview:

      System Version: macOS 14.4 (23E214)
      Kernel Version: Darwin 23.4.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: Damien’s MacBook Pro
      User Name: Damien Ostler (damienostler)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 1 day, 9 hours, 18 minutes
      
    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: Mac14,9
      Model Number: Z17H000CGLL/A
      Chip: Apple M2 Pro
      Total Number of Cores: 12 (8 performance and 4 efficiency)
      Memory: 32 GB
      System Firmware Version: 10151.101.3
      OS Loader Version: 10151.101.3
      Serial Number (system): K********C
      Hardware UUID: F********-****-****-****-***********E
      Provisioning UDID: 0*******-***************E
      Activation Lock Status: Disabled
```

Here are the commands I ran.

`brew install autoconf autoconf-archive automake ccache cmake ffmpeg nasm ninja pkg-config`
`brew install llvm`
`brew install qt`
`git clone https://github.com/LadybirdBrowser/ladybird.git`
`cd ladybird`
`CC=$(brew --prefix llvm)/bin/clang CXX=$(brew --prefix llvm)/bin/clang++ ./Meta/ladybird.sh run`

Here is the logs that I could get from the terminal.

https://hastebin.com/share/gelecitudi.bash

**Comment by sideshowbarker** [comment]
Looking at the tail of the log, I see the specific failure reported there is this:

> [563/2624] Linking CXX shared library lib/liblagom-gfx.0.0.0.dylib
> FAILED: lib/liblagom-gfx.0.0.0.dylib 
> …
> Undefined symbols for architecture arm64:
>   "___arm_tpidr2_save", referenced from:
>       _TransposeWxH_SME in libyuv.a[51](rotate_sme.cc.o)
>       _TransposeUVWxH_SME in libyuv.a[51](rotate_sme.cc.o)
> ld: symbol(s) not found for architecture arm64
> clang++: error: linker command failed with exit code 1 (use -v to see invocation)
>

**Comment by tobiaskohlbau** [comment]
Most likely a duplicate of #1526 which is addressed by #1580. I can confirm that using that PR branch does not show the libyuv error anymore.

**Comment by trflynn89** [comment]
Dup of #1526, fixed by #1580
